### PR TITLE
Now Different color can be given to Smilies.

### PR DESCRIPTION
**Pull request by harsh159357**
Now Different color can be given to Smilies. Fixed Issue No.9 sujithkanna#9 (distinct smilies color #9). But it will still need some more tweaking like setting color in while declaring SmileyRating in layout file

### CODE REVIEW

File: smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java [hunk 1]
```diff
@@ -311,31 +317,33 @@ protected void onDraw(Canvas canvas) {
...
+//                mDummyDrawPah.reset();
+//                mDummyDrawPah.addPath(mSmilePath, mScaleMatrix);
+//
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y,
+//                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
+//                canvas.drawPath(mDummyDrawPah, mPathPaint);
+//            } else {
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
+//                canvas.drawPath(mSmilePath, mPathPaint);
+//            }
```
Comment (addisonElliott) [inline smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:344]: Why did you comment out the `if (mSmileyNotSelectedPreviously)` code block?

It took me a little bit of digging when I was contributing to figure out what it does, so let me share my knowledge to you (if you don't already know). Basically, you want to provide a nice animation when you click on the smileys. When you have a selected smiley and you select a different one, it moves the smiley from the previous value to the new value.

However, if a smiley is not selected, then there is nowhere to start from. Thus, the variable mSmileyNotSelectedPreviously is used. Instead, it basically fades in/out the smiley instead.

File: smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java [hunk 2]
```diff
@@ -36,6 +35,13 @@
     private int mAngryColor = Color.parseColor("#f29a68");
     private int mNormalColor = Color.parseColor("#f2dd68");
```
Comment (addisonElliott) [inline smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:36]: There is no need for mAngryColor or even mNormalColor when you add colors for each of the smiley faces.

File: smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java [hunk 3]
```diff
@@ -311,31 +317,33 @@ protected void onDraw(Canvas canvas) {
...
+//                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
+//                canvas.drawPath(mDummyDrawPah, mPathPaint);
+//            } else {
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
+//                canvas.drawPath(mSmilePath, mPathPaint);
+//            }
+            canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
+            canvas.drawPath(mSmilePath, mPathPaint);
         }
     }
```
Comment (addisonElliott) [inline smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:349]: Should probably add option to set XML attributes for each of the smiley colors as well as functions setBadSmiley color.

**Review by addisonElliott — CHANGES_REQUESTED** [review]
Definitely a good start on this. It is a much needed feature but there are still some changes to be made.

Taking even one step further, I think it would be nice to have a configurable amount of smileys with independent colors.

File: smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java (hunk 2 above)
Comment (harsh159357) [reply smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:36]: Got your point. I left it out as it is so that what i have added can be easily reflected in code

File: smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java (hunk 1 above)
Comment (harsh159357) [reply smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:344]: I commented out this code block because there is a big problem inside this. Let Me explain you

1. Let's ignore my changes for this commented code-block only. Revert back this commented code block to the original one and run the app using that code.
2. Now tap on any smiley (first time) and then release it and after that try to swipe over the smilies you will see that the smilies colors aren't changing while swiping. And When you tap on any smiley (second time) and then swipe over the smilies you will see that the smilies colors are changing.

Basically, in short On First Tap and then Swipe over smilies the color change is not working for independent smilies. Even in the case of current stable master branch

Comment (harsh159357) [reply smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:344]: More explanation with steps -

1. Just Uncomment Out the code block (line 320 - 344) which I have commented out and comment out(line 345-346)
2. Say for an example tap on OKAY smiley or any (Now okay smiley selected and its color is yellowish. Also, this is our first time when we have clicked on our SmileyRating Widget). Now just swipe over the smilies by holding the smiley which is selected by tap i.r OKAY smiley (you will be able to observe strange behavior that smilies color isn't changing at all)
3. After Step 2 release your swipe and then follow step 2 again you will see that while swiping smilies color is changing that's it.
4. So for that strange thing, I have used the following code block
            canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
            canvas.drawPath(mSmilePath, mPathPaint);

So this thing solves color change for smiles for the first swipe.

Comment (addisonElliott) [reply smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java:344]: Interesting, I don't have a lot of time to download the request and run it, but I **think** I see what you mean. 

I may know what the issue is. If you look at lines 326-328 in the code you commented out, you can see that it selects between the color mAngryColor or mNormalColor. Did you try this with using the correct colors? (Instead, you do a switch statement and get the right color between the 5).

**Comment by sujithkanna** [comment]
I accepted the merge request. I will add the rest of the API's for this feature and will be released as a major update with few fixes. Thank you for your contribution @harsh159357

**Comment by harsh159357** [comment]
Nice to hear this good @sujithkanna that my pull request is merged. Thanks for creating this wonderful library.
